dashboard-app/src/app/pages/room-type/controllers/RoomCtrl.js

```diff
@@ -0,0 +1,55 @@
+(function () {
+    'use strict';
+    angular.module('ConciergeApp.pages.roomType')
+        .controller('RoomCtrl', RoomCtrl);
+
+    RoomCtrl.$inject = ['RoomTypeService', 'RoomService', '$scope', 'toastr'];
+    function RoomCtrl(RoomTypeService, RoomService, $scope, toastr) {
+
+        $scope.room = {};
+
+        RoomTypeService.getRoomTypes().then(function (response) {
+            $scope.roomTypes = response.data;
+        }, function (error) {
+            console.log(error);
+        });
+
+        $scope.submit = function (roomForm) {
+            if (roomForm.$invalid) {
+                return;
+            }
+            RoomService.saveRoom($scope.room).then(function (response) {
+                if (response.status === 200) {
+                    toastr.success('Room has been saved successfully', 'Save Room');
+                    roomForm.$setPristine();
+                    roomForm.$setUntouched();
+                    $scope.room = {};
+                    $scope.selectedRoomType = null;
+                } else {
+                    if (response.data.errors) {
+                        angular.forEach(response.data.errors, function (value, key) {
+                            toastr.error(response.data.errors[key].defaultMessage, 'Error');
+                        });
+                    } else {
+                        toastr.error(response.data.error, 'Error');
+                    }
+                }
+            }, function (error) {
+                console.log(error);
+            })
+        }
+
+        $scope.changeRoomType = function (roomType) {
+            $scope.selectedRoomType = roomType;
+            RoomTypeService.getImage(roomType.image).then(function (data) {
+                var canvas = document.getElementById("room-type-image");
```

Comment:
Use angular approach. Do we need to use canvas? You can use simple img tag with path to image instead of canvas. In that way you just need to set ng-src with value from scope.